source/cursorManager.py

```diff
@@ -386,8 +386,10 @@ def _selectionMovementScriptHelper(self,unit=None,direction=None,toPosition=None
 				# 1. Caret at 1; selection (1, 1)
 				# 2. Shift+rightArrow: selection (1, 2)
 				newInfo.setEndPoint(oldInfo, "startToStart")
-		self.selection = newInfo
-		speech.speakSelectionChange(oldInfo,newInfo)
+		# Update selection and speak change when selection has changed
+		if oldInfo != newInfo:
+			self.selection = newInfo
+			speech.speakSelectionChange(oldInfo, newInfo)

```

Comment:
@michaelDCurran - any thoughts on this change?

I think we should always be updating the selection here, but I can understand why we may avoid speaking selection change

Reply:
I think @michaelDCurran's ideas about this change would still be valuable here,